public/index.php

```diff
@@ -10,23 +10,36 @@
     throw new \ErrorException($message, 0, $severity, $file, $line);
 });
 
-$jsonRequest = $_SERVER['REQUEST_METHOD'] !== 'GET' || stripos($_SERVER['HTTP_ACCEPT'], 'application/json') !== false;
+//Switch to json if it was specified in the accept header.
+$jsonRequest = stripos($_SERVER['HTTP_ACCEPT'], 'application/json') !== false;
 
 try {
-    require_once __DIR__ . '/../vendor/autoload.php';
-
     if (!$jsonRequest) {
-        // add common head tags here. Browser should merge them with the ones in the view file.
+        // add common head tags here. Browser will merge them with any head tags printed afterwards.
         echo <<<HTML
             <head>
                 <link rel="icon" type="image/png" href="/favicon.png">
                 <link rel="stylesheet" href="/css/reset.css">
+                <link rel="stylesheet" href="/css/global.css">
```

Comment:
If added via a head tag this now means the global css will be cached and "keeping it to a minimmum" may no longer be a concern.
The main thing to worry about is if doing it this way will still prevent the dark mode flash.